pkg/resource/resource.go

```diff
@@ -212,6 +214,11 @@ func IsAPIError(err error) bool {
 	return ok
 }
 
+// IsAPIErrorWrapped returns true if err is a K8s API error, or recursively wraps a K8s API error
```

Comment:
From my reading of the source code, it doesn't seem like we'd get more than one level below, i.e. if it's wrapped two times, we won't actually see the first error because `Wrap` treats the incoming error as cause, not the cause of that incoming error.

If that's the case, I wonder if we should make this APIError check closer to where the error is returned without relying on whether the wrapping is done only once.

Reply:
`resource.IsAPIErrorWrapped` uses `errors.Cause` which recursively looks for the first non-causer error. So, even if the original error is wrapped more than once, we are safe to get the original non-causer error.

Reply:
Got it, seems I misinterpreted the source code. Thanks for the explanation!